### PR TITLE
docs(dotnet): examples for events, handles

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/src/events.md [hunk 1]
```diff
@@ -140,6 +158,19 @@ page.remove_listener("requestfinished", print_request_finished)
 page.goto("https://www.openstreetmap.org/")
 ```
 
+```csharp
+page.Request += (_, request) => Console.WriteLine("Request sent: " + request.Url);
+EventHandler<IRequest> listener = (_, request) => {
```
Comment (kblok) [inline docs/src/events.md:163]: Try this, please.

```suggestion
void listener(object sender, IRequest request) => 
{
```

File: docs/src/handles.md [hunk 2]
```diff
@@ -126,6 +136,20 @@ class_names = element_handle.get_attribute('class')
 assert 'highlighted' in class_names
 ```
 
+```csharp
+// Get the element handle
+var jsHandle = await page.WaitForSelectorAsync("#box");
+var elementHandle = jsHandle as ElementHandle;
```
Comment (kblok) [inline docs/src/handles.md:142]: wondering about this. If we keep the `AsElementHandle` function, why not using that?
I like more the example version tho.

File: docs/src/handles.md [hunk 3]
```diff
@@ -213,6 +237,25 @@ page.evaluate("(arg) => arg.myArray.push(arg.newElement)", {
 my_array_handle.dispose()
 ```
 
+```csharp
+// Create new array in page.
+var myArrayHandle = await page.EvaluateHandleAsync("() => {\n" +
+  "  window.myArray = [1];\n" +
+  "  return myArray;\n" +
+  "}");
```
Comment (kblok) [inline docs/src/handles.md:245]:
```suggestion
var myArrayHandle = await page.EvaluateHandleAsync(@"() => {
    window.myArray = [1];
    return myArray;
}");
```

File: docs/src/events.md [hunk 4]
```diff
@@ -49,6 +49,15 @@ with page.expect_request("**/*logo*.png") as first:
 print(first.value.url)
 ```
 
+```csharp
+// The callback lambda defines scope of the code that is expected to
```
Comment (dgozman) [inline docs/src/events.md:53]: This comment does not apply.

File: docs/src/events.md [hunk 5]
```diff
@@ -84,6 +93,15 @@ with page.expect_popup() as popup:
 popup.value.goto("https://wikipedia.org")
 ```
 
+```csharp
+// The callback lambda defines scope of the code that is expected to
```
Comment (dgozman) [inline docs/src/events.md:97]: Same here

File: docs/src/events.md [hunk 6]
```diff
@@ -140,6 +158,19 @@ page.remove_listener("requestfinished", print_request_finished)
...
 
+```csharp
+page.Request += (_, request) => Console.WriteLine("Request sent: " + request.Url);
+EventHandler<IRequest> listener = (_, request) => {
+  Console.WriteLine("Request finished: " + request.Url);
+};
+page.RequestFinished += listener;
+await page.GotoAsync("https://wikipedia.org");
+
+// Remove previously added listener, each on* method has corresponding off*
```
Comment (dgozman) [inline docs/src/events.md:169]: `each on* method has corresponding off*` does not apply

File: docs/src/events.md [hunk 7]
```diff
@@ -164,6 +195,11 @@ page.once("dialog", lambda dialog: dialog.accept("2021"))
 page.evaluate("prompt('Enter a number:')")
 ```
 
+```csharp
+page.Dialog += (_, dialog) => dialog.AcceptAsync("2021");
```
Comment (dgozman) [inline docs/src/events.md:199]: This is a section about once listener - do we even have that in C#?

File: docs/src/handles.md [hunk 8]
```diff
@@ -213,6 +237,25 @@ page.evaluate("(arg) => arg.myArray.push(arg.newElement)", {
...
+
+// Get the length of the array.
+var length = await page.EvaluateAsync<int>("a => a.length", myArrayHandle);
+
+// Add one more element to the array using the handle
+await page.EvaluateAsync("arg => arg.myArray.add(arg.newElement)",
+    new { myArray = myArrayHandle, newElement = 2 });
+
+// Release the object when it"s no longer needed.
```
Comment (dgozman) [inline docs/src/handles.md:254]: typo:  it"s

File: docs/src/events.md (hunk 7 above)
Comment (kblok) [reply docs/src/events.md:199]: We don't

**Pull request merged**